src/dotnet/UI.Blazor/Components/VisualMediaViewerModal/visual-media-viewer.ts

```diff
@@ -200,6 +309,173 @@ export class VisualMediaViewer {
 
     private onPointerUp = (event: PointerEvent) => {
         window.removeEventListener('pointermove', this.onPointerMove);
+        if (event.pointerType === 'touch') {
+            this.removeEvent(event);
```

Comment:
This kinda sucks. IMO event removal should be extremely reliable - i.e. somewhere in a common piece of logic.

Same for the subscription.